src/js/auth.js

```diff
@@ -0,0 +1,72 @@
+import { UserManager, WebStorageStateStore } from 'oidc-client-ts';
+import crypto from 'crypto-js';
+
+const settings = {
+  authority: 'https://idp.int.identitysandbox.gov',
+  client_id: 'urn:gov:gsa:openidconnect.profiles:sp:sso:gsa:gsa-fac-pkce-01',
+  redirect_uri: window.location.origin + '/auth/post-login',
```

Comment:
@heymatthenry @jasnakai - currently unable to make use of the federalist preview, because the base url for the site is `*app.cloud.gov/preview/foo/bar`. This line fails to build the correct redirect uri in that scenario, because it references the origin (`*.app.cloud.gov`) instead of the actual base url, which includes some additional paths. The redirect uri in a federalist preview context works out to `*.app.cloud.gov/auth/post-login`, which isn't correct.

Is there a better way to accomplish what I'm attempting here, which would work in both cases? Or is this a problem possibly not worth spending time solving?

Reply:
@timoballard Is the [`BASEURL`](https://github.com/GSA-TTS/FAC-Frontend/blob/main/src/_data/config.js#L1) environment variable useful here? We have to use that for asset paths because of Federalist's funky branch-name-based URLs.

Reply:
[Relevant Federalist docs](https://federalist.18f.gov/documentation/env-vars-on-federalist-builds/)

Reply:
@heymatthenry - yes, that appears to be what I needed - thank you! After... 4 commits' worth of fumbling around, things appear to work now in local, testing, and federalist-preview contexts 🎉